### PR TITLE
Support new entity memo fields

### LINKED ISSUE #1553: Support new entity memo fields
**Problem**
HAPI is adding new entity memo fields for accounts, files and tokens and the mirror node should support them.

**Solution**
- Populate the `t_entities.memo` field for any new memo fields

**Alternatives**

**Additional Context**

https://github.com/hashgraph/hedera-services/issues/962
https://github.com/hashgraph/hedera-services/issues/961
https://github.com/hashgraph/hedera-services/issues/960

### PR DESCRIPTION
**Pull request by xin-hedera**
**Detailed description**:

- Support new / updated memo fields in ContractUpdate, CryptoCreate/Update, FileCreate/Update, and TokenCreate/Update transactions
- Fix bug in EntityRecordItemListenerTest for token and schedule

**Which issue(s) this PR fixes**:
Fixes #1553 

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/record/transactionhandler/AbstractUpdatesEntityTransactionHandlerTest.java [hunk 1]
```diff
@@ -1,6 +1,6 @@
 package com.hedera.mirror.importer.parser.record.transactionhandler;
 
-/*
+/*-
  * ‌
  * Hedera Mirror Node
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/record/transactionhandler/AbstractUpdatesEntityTransactionHandlerTest.java:5]: For each of the update handlers with a memo, we should add a common test case that clears the memo to empty string and one that doesn't set either. The purpose of the `StringValue` wrapper is to differentiate between unset, set and clear.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/record/transactionhandler/AbstractUpdatesEntityTransactionHandlerTest.java [hunk 2]
```diff
@@ -1,6 +1,6 @@
 package com.hedera.mirror.importer.parser.record.transactionhandler;
 
-/*
+/*-
```
[the remark this answers is not in the document]
Comment (ijungmann) [reply hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/record/transactionhandler/AbstractUpdatesEntityTransactionHandlerTest.java:3]: Isn't that dash on all the licenses?

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/ContractCreateTransactionHandler.java [hunk 3]
```diff
@@ -59,12 +59,11 @@ public void updateEntity(Entities entity, RecordItem recordItem) {
...
-        // Can't clear memo on contracts. 0 length indicates no change
-        if (txMessage.getMemo() != null && txMessage.getMemo().length() > 0) {
-            entity.setMemo(txMessage.getMemo());
-        }
+
         if (txMessage.hasAdminKey()) {
             entity.setKey(txMessage.getAdminKey().toByteArray());
         }
+
+        entity.setMemo(txMessage.getMemo());
```
Comment (xin-hedera) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/ContractCreateTransactionHandler.java:67]: fix bug in `ContractCreateTransactionHandler`, when the memo `String` is either not set or set to empty string, the memo in the entity should be set to empty string

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/record/transactionhandler/AbstractTransactionHandlerTest.java [hunk 4]
```diff
@@ -68,6 +110,47 @@ void testGetEntityId() {
                 expectedEntityId);
     }
 
+    @TestFactory
+    Stream<DynamicTest> testUpdateEntity() {
```
Comment (ijungmann) [inline hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/record/transactionhandler/AbstractTransactionHandlerTest.java:114]: Clever test.

**Review by ijungmann — APPROVED** [review]
LGTM

**Review by Nana-EC — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]
LGTM
